### PR TITLE
[Merged by Bors] - chore(Data/Finset): drop some `Nonempty` arguments

### PR DESCRIPTION
**Pull request by urkud**
* rename `Finset.Nonempty.image_iff` to `Finset.image_nonempty`, deprecate the old version;
* rename `Set.nonempty_image_iff` to `Set.image_nonempty`, deprecate the old version;
* drop unneeded `Finset.Nonempty` arguments here and there;
* add versions of some lemmas that assume `Nonempty s` instead of `Nonempty (s.image f)` or `Nonempty (s.map f)`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Image.lean [hunk 1]
```diff
@@ -408,22 +408,24 @@ theorem fiber_nonempty_iff_mem_image (f : α → β) (s : Finset α) (y : β) :
 #align finset.fiber_nonempty_iff_mem_image Finset.fiber_nonempty_iff_mem_image
 
 @[simp, norm_cast]
-theorem coe_image {f : α → β} : ↑(s.image f) = f '' ↑s :=
+theorem coe_image : ↑(s.image f) = f '' ↑s :=
   Set.ext <| by simp only [mem_coe, mem_image, Set.mem_image, implies_true]
 #align finset.coe_image Finset.coe_image
 
+@[simp]
+lemma nonempty_image_iff : (s.image f).Nonempty ↔ s.Nonempty := by
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Image.lean:416]: While you're at it, the finset naming convention would rather choose
```suggestion
lemma image_nonempty : (s.image f).Nonempty ↔ s.Nonempty := by
```

File: Mathlib/Data/Finset/Lattice.lean [hunk 2]
```diff
@@ -1090,9 +1087,9 @@ theorem _root_.map_finset_inf' [SemilatticeInf β] [InfHomClass F α β] (f : F)
 
 @[simp]
 theorem inf'_image [DecidableEq β] {s : Finset γ} {f : γ → β} (hs : (s.image f).Nonempty)
-    (g : β → α) (hs' : s.Nonempty := (Nonempty.image_iff _).1 hs) :
-    (s.image f).inf' hs g = s.inf' hs' (g ∘ f) :=
-  @sup'_image αᵒᵈ _ _ _ _ _ _ hs _ hs'
+    (g : β → α)  :
+    (s.image f).inf' hs g = s.inf' hs.of_image (g ∘ f) :=
+  @sup'_image αᵒᵈ _ _ _ _ _ _ hs _
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Lattice.lean:1092]: For rewriting, can we have the same lemma but turned around and with `(hs : s.Nonempty)`?

File: Mathlib/Data/Finset/Image.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Finset/Image.lean:416]: I aligned the name with `Set.nonempty_image_iff`.

File: Mathlib/Data/Finset/Lattice.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Data/Finset/Lattice.lean:1094]: Done. I hope, I didn't miss anything.

File: Mathlib/Data/Finset/Image.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Finset/Image.lean:416]: Yes, but `Finset` and `Set` differ in naming. FWIW I am of the opinion that it should be `Set.image_nonempty` too.

Comment (urkud) [reply Mathlib/Data/Finset/Image.lean]: OK, let's try this.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**